ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/BuilderBidValidatorImpl.java

```diff
@@ -122,20 +122,21 @@ public void validateBuilderBid(
     final UInt64 preferredGasLimit = validatorRegistration.getGasLimit();
     final UInt64 proposedGasLimit = executionPayloadHeader.getGasLimit();
 
-    if (parentGasLimit.equals(preferredGasLimit) && proposedGasLimit.equals(parentGasLimit)) {
-      return;
-    }
-
-    if (preferredGasLimit.isGreaterThan(parentGasLimit)
-        && proposedGasLimit.isGreaterThan(parentGasLimit)) {
-      return;
+    final UInt64 expectedGasLimit = expectedGasLimit(parentGasLimit, preferredGasLimit);
+    if (!expectedGasLimit.equals(preferredGasLimit)) {
+      eventLogger.builderBidNotHonouringGasLimit(
+          parentGasLimit, proposedGasLimit, preferredGasLimit);
     }
+  }
 
-    if (preferredGasLimit.isLessThan(parentGasLimit)
-        && proposedGasLimit.isLessThan(parentGasLimit)) {
-      return;
+  static UInt64 expectedGasLimit(final UInt64 parentGasLimit, final UInt64 targetGasLimit) {
+    final UInt64 maxGasLimitDifference = parentGasLimit.dividedBy(1024L).minusMinZero(1);
```

Comment:
Why is the `.minusMinZero(1)` here?

Ah. It's because it asserts that it's strictly less than or greater than the value.

<img width="989" alt="image" src="https://github.com/user-attachments/assets/30ab812c-b4c5-4899-b454-54fb78cce905" />

For example, given that

* `parent_gas_limit == 1024_000`
* `parent_gas_limit // 1024 == 1000`

`gas_limit` must be less than `1024_000 + 1000` (1025_000), so the max value is 1024_999.

Reply:
i think given the thread and the test match, i'll mark this resolved but feel free to ping me if this is an incorrect assertion

Reply:
> i'll try to find the original spec entry, i was referencing a PR from prysm, which did the -1...

so I ended up here while looking for this in the spec, only to notice it actually is not part of any spec but only documented in EIP-1559...

@jtraglia should we add this to the bellatrix builder spec?